src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+
+  actions.forEach(action => {
+    let result;
+
+    if (!results.length) {
+      result = {
+        ...state,
+      };
+    } else {
+      result = { ...results[results.length - 1] };
+    }
```

Comment:
```suggestion
    let result = { ...state }
```
Для чого так ускладнювати? Просто створи копію стейту ПОЗА межами циклу, і далі всередині циклу працюй з цією копіє (краще назвати цю змінну не result, а stateCopy)

Reply:
Я спробував переписати код за вашими рекомендаціями, але я вважаю що моя  конструкція потрібна для збереження покрокових змін стейту, як того потребують тести!